server/broadcast_test.go

```diff
@@ -247,15 +250,18 @@ func TestSelectSession(t *testing.T) {
 
 	// assert last session selected and sessList is correct length
 	sess := bsm.selectSession()
+	time.Sleep(2 * time.Millisecond)
```

Comment:
Are these time.Sleep calls necessary? Looks like the next few tests aren't checking for refresh behavior.

Reply:
I added these sleeps because the previous goroutine wasn’t finishing before starting the waitgroup test, the wait group counter got deducted by the goroutine from the previous test and so our actual test would result in a negative count and thus a panic. This happened consistently for me locally.

Reply:
Ah I see.

I think another way to resolve this without using sleeps is to create a new `bsm` before the waitgroup test. Starting after L270:

```golang
bsm = newSessionsManagerLIFO(bsmWithSessList([]*BroadcastSession{}))
bsm.sessMap["transcoder1"] = StubBroadcastSession("transcoder1")
bsm.sessMap["transcoder2"] = StubBroadcastSession("transcoder2")
```

The last 2 lines aren't strictly necessary. But, previously, sessMap had 2 entries when running the waitgroup test
and we can replicate that scenario with the last 2 lines.

As mentioned in [this comment](https://github.com/livepeer/go-livepeer/issues/1370#issuecomment-611250411) I think it is worthwhile to avoid introducing new tests that use sleeps if we can given the flaky behavior we've seen in CI in the past.

Reply:
Since we also alter the `createSessions` implementation for this specific test I don't really see much added value in trying to necessarily replicate the previous scenario as it wasn't really relevant initially either. But I'm okay with creating a new `bsm` , thanks !

Reply:
fixed in d313eb53